CRAMv3.tex

```diff
@@ -732,9 +736,19 @@ \subsection{\textbf{Slice header block}}
 
 The slice header block is never compressed (block method=raw). For reference mapped 
 reads the slice header also defines the reference sequence context of the data 
-blocks associated with the slice. Mapped and unmapped reads can be stored within 
-the same slice similarly to BAM file. Slices with unsorted reads must not contain 
-any other types of reads.
+blocks associated with the slice. Mapped reads can be stored along with
+\textbf{placed unmapped}\footnote{By placed unmapped read we mean a read that is unmapped according to 
+its BAM bit flags but has position fields filled in, thus "placing"  it on a reference sequence. In contrast, 
+unplaced unmapped reads have have a reference sequence ID of -1 and alignment position of 0.}
+reads on the same reference within the same slice, as is possible in BAM.  
+Slices may also contain reads mapped to
+different external references, by setting the Multiple Reference flag (-2) as the sequence ID in the header.
+These can also include placed unmapped reads on these references, but multiple embedded references 
+cannot be combined in this way.  When multiple references are used, the RI data series will be used to 
+determine the reference sequence ID for each record. This data series is not required when only a single 
+reference is used within a slice.   Slices with unplaced unmapped\footnotemark[\value{footnote}] reads 
```

Comment:
I think this is wrong and I'm over-constraining here.

This was based on an interpretation of the old text "Slices with unsorted reads must not contain any other types of reads."  What is an "unsorted read" independently of the context of other reads?  No, it's the overall file header which determines sortedness.  For any sorting besides Coordinate, Slices and Containers will often have reads with a variety of references, so Multi-Ref (-2) by necessity.  What would it gain us to exclude Unmapped-Unsorted reads from these slices?  It could be quite bad in the pathological case: (mapped, unmapped, mapped, unmapped, mapped) would require a new slice at every record boundary!

So I'd like to remove this.  What do you think?

Reply:
Added this change as commit bf44654

Reply:
I agree in name sorted data it could be pathologically bad.

I'm not sure what we do in this scenario, but it's probably quite likely we just mix them in the same slice.  I don't see any issue with stating multi-ref (-2) and having ref -1 (unmapped) for some of the reads.  It feels natural.  This text is probably just a hangover from when we had multiple slice types.